### PR TITLE
add genericSchemeRegistry with en/decoding generic schemes

### PR DESCRIPTION
**Pull request by jellegerbrandy**

### CODE REVIEW

File: src/schemes/genericSchemeRegistry.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  name: string
+  inputs: Array<{ name: string, type: string}>
+  outputs: any[]
+  payable: boolean
+  stateMutability: string
+  type: string
+}
+export interface IActionSpec {
+  id: string
+  label: string
```
Comment (tibetsprague) [inline src/schemes/genericSchemeRegistry.ts]: we will also want placeholderText, and probably a defaultValue

File: src/schemes/genericSchemeRegistry.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  public specs: IGenericSchemeJSON
+
+  constructor(info: IGenericSchemeJSON) {
+    this.specs = info
+  }
+  public actions() {
+    return this.specs.actions
+  }
+  public action(id: string) {
+    for (const action of this.specs.actions) {
```
Comment (tibetsprague) [inline src/schemes/genericSchemeRegistry.ts]: you could use the array.find() function here

File: src/schemes/genericSchemeRegistry.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      if (action.id === id) {
+        return action
+      }
+    }
+    throw Error(`An action with id ${id} coult not be found`)
+  }
+  public actionByFunctionName(name: string) {
+    for (const action of this.specs.actions) {
```
Comment (tibetsprague) [inline src/schemes/genericSchemeRegistry.ts]: .find here too

File: src/schemes/genericSchemeSpecs/dxDao.json [hunk 4]
```diff
@@ -0,0 +1,128 @@
+{
```
Comment (tibetsprague) [inline src/schemes/genericSchemeSpecs/dxDao.json]: missing the fields section?

**Comment by jellegerbrandy** [comment]
this is part of alchemy for the moment, and we'll keep it there until the code is more stable

**Pull request closed without merging**